app/services/request_create_service.rb

```diff
@@ -21,15 +21,23 @@ def create(options)
       :stages   => stages
     )
     Request.create!(create_options).tap do |request|
-      start_approval_process(request) if ENV['AUTO_APPROVAL'] && ENV['AUTO_APPROVAL'] != 'n'
+      start_approval_process(request) if default_approve? || auto_approve?
     end
   end
 
   private
 
+  def default_approve?
+    workflow.id == Workflow.default_workflow.try(:id)
+  end
+
+  def auto_approve?
+    ENV['AUTO_APPROVAL'] && ENV['AUTO_APPROVAL'] != 'n'
+  end
+
   def start_approval_process(request)
     Thread.new do
-      auto_approve(request)
+      default_approve? ? default_approve(request) : auto_approve(request)
```

Comment:
Question about `default_approve`: what's its purpose? Seems like `auto_approve` is more similar to the real case.

Reply:
I changed the seeded workflow from `auto_approve` to `always_approve` and use it as a default. As indicated by the name this workflow always approves unconditionally; no approver groups are needed.

The concept of "auto approve" is reserved for the use of algorithm approval based on business rules. The result could be approved or denied. Currently `auto_approve` controlled by environmental variables is to simulate multi-level approvals. It will not be used in production.

Reply:
👍